language-server/src/dotty/tools/languageserver/DottyLanguageServer.scala

```diff
@@ -65,7 +66,8 @@ class DottyLanguageServer extends LanguageServer
       myDrivers = new mutable.HashMap
       for (config <- configs) {
         val classpathFlags = List("-classpath", (config.classDirectory +: config.dependencyClasspath).mkString(File.pathSeparator))
-        val settings = defaultFlags ++ config.compilerArguments.toList ++ classpathFlags
+        val sourcepathFlags = List("-sourcepath", config.sourceDirectories.mkString(File.pathSeparator), "-scansource")
+        val settings = defaultFlags ++ config.compilerArguments.toList ++ classpathFlags ++ sourcepathFlags
```

Comment:
When running launchIDE on Dotty we get:
> [error] Flag -sourcepath set repeatedly

Because `-sourcepath` is part of the settings we set for dotty-library in our build. This could be solved by not adding `-sourcepath` when it's already set, or by appending source directories to an existing `-sourcepath` found in `config.compilerArguments`

Reply:
Not sure what is the right thing to do here. I was assuming config.courseDirectories gives us all the info. Does it make sense to include the dotty-library sourcepath?

Reply:
sourceDirectories does give us all the info, it's just that when compiling dotty-library itself we already use -sourcepath in our build to avoid accidentally linking against the dotty-library on the classpath. In general we cannot assume that -sourcepath is not part of `compilerArguments`.

Reply:
Ah, OK, so we can just filter out the existing sourcepath, right?